python/dask_cudf/dask_cudf/expr/_collection.py

```diff
@@ -71,6 +81,20 @@ def groupby(
                 f"`by` must be a column name or list of columns, got {by}."
             )
 
+        if "as_index" in kwargs:
+            warnings.warn(
+                "The `as_index` argument is no longer supported in "
+                "dask-cudf when query-planning is enabled.",
+                FutureWarning,
+            )
+
+        if kwargs.pop("as_index", True) is not True:
+            raise NotImplementedError(
+                f"`as_index=False` is not supported. Please disable "
+                "query planning, or reset the index after aggregating.\n"
+                f"{_LEGACY_WORKAROUND}"
+            )
```

Comment:
This essentially does what @charlesbluca suggested - The message is still slightly confusing, but so is the behavior I guess.